### ISSUE TITLE
Error: Could not get the Java version. Is Java installed?

### ISSUE THREAD

**Issue by MiteshShah** [appium/appium#6044]
```
$ appium
info: Welcome to Appium v1.4.16 (REV ae6877eff263066b26328d457bd285c0cc62430d)
info: Appium REST http interface listener started on 0.0.0.0:4723
info: Console LogLevel: debug
info: --> POST /wd/hub/session {"desiredCapabilities":{"appPackage":"com.womistudios.arcadetap.aphone","appActivity":"com.womistudios.arcadetap.aphone.Splashscreen","platformName":"Android","udid":"10.0.0.9","deviceName":"S8WCV8UWGUJ7MVNF"}}
info: Client User-Agent string: Apache-HttpClient/4.5.1 (Java/1.8.0_45-internal)
info: [debug] Didn't get app but did get Android package, will attempt to launch it on the device
info: [debug] Creating new appium session e0bf11a2-06b5-49d9-8790-75c04768efc0
info: Starting android appium
info: [debug] Getting Java version
info: [debug] Cleaning up android objects
info: [debug] Cleaning up appium session
error: Failed to start an Appium session, err was: Error: Could not get the Java version. Is Java installed?
info: [debug] Error: Could not get the Java version. Is Java installed?
    at [object Object].<anonymous> (/home/perk/node-v4.2.1-linux-x64/lib/node_modules/appium/lib/devices/android/android-common.js:1062:17)
    at ChildProcess.exithandler (child_process.js:194:7)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:818:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
info: [debug] Responding to client with error: {"status":33,"value":{"message":"A new session could not be created. (Original error: Could not get the Java version. Is Java installed?)","origValue":"Could not get the Java version. Is Java installed?"},"sessionId":null}
info: <-- POST /wd/hub/session 500 91.143 ms - 222
```

```
$ java --version
Unrecognized option: --version
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```
```
$ java -version
openjdk version "1.8.0_45-internal"
OpenJDK Runtime Environment (build 1.8.0_45-internal-b14)
OpenJDK 64-Bit Server VM (build 25.45-b02, mixed mode) 
```

**Comment by MiteshShah** [comment]
```
$ appium-doctor 
Running Android Checks
✔ ANDROID_HOME is set to "/home/mitesh/Android/Sdk/"
✔ JAVA_HOME is set to "/usr/lib/jvm/java-8-openjdk-amd64/jre/."
✔ ADB exists at /home/mitesh/Android/Sdk/platform-tools/adb
✔ Android exists at /home/mitesh/Android/Sdk/tools/android
✔ Emulator exists at /home/mitesh/Android/Sdk/tools/emulator
✔ Android Checks were successful.

✔ All Checks were successful
```

**Comment by MiteshShah** [comment]
Issue is fixed by #6045
